src/Microsoft.AspNetCore.Mvc.Razor/ViewComponentTagHelpers/ViewComponentTagHelperDescriptorFactory.cs

```diff
@@ -0,0 +1,162 @@
+        /// <param name="descriptorProvider">The provider of <see cref="ViewComponentDescriptor"/>s.</param>
+        public ViewComponentTagHelperDescriptorFactory(IViewComponentDescriptorProvider descriptorProvider)
+        {
+            if (descriptorProvider == null)
+            {
+                throw new ArgumentNullException();
+            }
+
+            _descriptorProvider = descriptorProvider;
+        }
+
+        /// <summary>
+        /// Creates <see cref="TagHelperDescriptor"/> representations of <see cref="ViewComponents"/>
+        /// in an <see href="Assembly"/> represented by the given <paramref name="assemblyName"/>.
+        /// </summary>
+        /// <param name="assemblyName">The name of the assembly containing 
+        /// the <see cref="ViewComponents"/> s to translate.</param>
+        /// <returns>A <see cref="IEnumerable{TagHelperDescriptor}"/>, 
+        /// one for each <see cref="ViewComponents"/> .</returns>
+        public IEnumerable<TagHelperDescriptor> CreateDescriptors(string assemblyName)
+        {
+            if (assemblyName == null)
+            {
+                throw new ArgumentNullException(assemblyName);
+            }
+
+            var viewComponentDescriptors = _descriptorProvider
+                .GetViewComponents()
+                .Where(viewComponent => assemblyName.Equals(
+                    viewComponent.TypeInfo.Assembly.GetName().Name, StringComparison.Ordinal));
+
+            var tagHelperDescriptors = CreateDescriptors(viewComponentDescriptors);
+            return tagHelperDescriptors;
+        }
+
+        /*
+        private IEnumerable<TagHelperDescriptor> CreateDescriptors(
+            IEnumerable<ViewComponentDescriptor> viewComponentDescriptors) =>
+            viewComponentDescriptors.Select(viewComponentDescriptor => CreateDescriptor(viewComponentDescriptor));
+            */
+
+        private IEnumerable<TagHelperDescriptor> CreateDescriptors(IEnumerable<ViewComponentDescriptor> viewComponentDescriptors)
+        {
+            var tagHelperDescriptors = new List<TagHelperDescriptor>();
+
+            foreach (var viewComponent in viewComponentDescriptors)
+            {
+                var descriptor = CreateDescriptor(viewComponent);
+                tagHelperDescriptors.Add(descriptor);
+            }
+
+            return tagHelperDescriptors;
+        }
+        private TagHelperDescriptor CreateDescriptor(ViewComponentDescriptor viewComponentDescriptor)
+        {
+            var assemblyName = viewComponentDescriptor.TypeInfo.Assembly.GetName().Name;
+            var tagName = GetTagName(viewComponentDescriptor);
+            var typeName = GetTypeName(viewComponentDescriptor);
+
+            var tagHelperDescriptor = new TagHelperDescriptor
+            {
+                TagName = tagName,
+                TypeName = typeName,
+                AssemblyName = assemblyName,
+                TagStructure = TagStructure.NormalOrSelfClosing,
```

Comment:
Hmm, thinking back on this guy, lets just not set this to let it be unspecified.